cpp/src/decisiontree/memory.h

```diff
@@ -23,45 +23,99 @@
 template <class T, class L>
 struct TemporaryMemory {
   // Labels after boostrapping
-  MLCommon::device_buffer<L> *sampledlabels;
+  MLCommon::device_buffer<L> *sampledlabels = nullptr;
 
   // Used for gini histograms (root tree node)
-  MLCommon::device_buffer<int> *d_hist;
-  MLCommon::host_buffer<int> *h_hist;
+  MLCommon::device_buffer<int> *d_hist = nullptr;
+  MLCommon::host_buffer<int> *h_hist = nullptr;
 
   //Host/Device histograms and device minmaxs
-  MLCommon::device_buffer<T> *d_globalminmax;
-  MLCommon::device_buffer<int> *d_histout;
-  MLCommon::device_buffer<unsigned int> *d_colids;
-  MLCommon::host_buffer<int> *h_histout;
-  MLCommon::device_buffer<T> *d_mseout, *d_predout;
-  MLCommon::host_buffer<T> *h_mseout, *h_predout;
+  MLCommon::device_buffer<T> *d_globalminmax = nullptr;
+  MLCommon::device_buffer<int> *d_histout = nullptr;
+  MLCommon::host_buffer<int> *h_histout = nullptr;
+  MLCommon::device_buffer<T> *d_mseout = nullptr;
+  MLCommon::device_buffer<T> *d_predout = nullptr;
+  MLCommon::host_buffer<T> *h_mseout = nullptr;
+  MLCommon::host_buffer<T> *h_predout = nullptr;
 
   //Below pointers are shared for split functions
-  MLCommon::device_buffer<char> *d_flags_left, *d_flags_right;
-  MLCommon::host_buffer<int> *nrowsleftright;
+  MLCommon::device_buffer<char> *d_flags_left = nullptr;
+  MLCommon::device_buffer<char> *d_flags_right = nullptr;
+  MLCommon::host_buffer<int> *nrowsleftright = nullptr;
   MLCommon::device_buffer<char> *d_split_temp_storage = nullptr;
   size_t split_temp_storage_bytes = 0;
 
-  MLCommon::device_buffer<int> *d_num_selected_out;
-  MLCommon::device_buffer<unsigned int> *temprowids;
-  MLCommon::device_buffer<T> *question_value, *temp_data;
+  MLCommon::device_buffer<int> *d_num_selected_out = nullptr;
+  MLCommon::device_buffer<unsigned int> *temprowids = nullptr;
+  MLCommon::device_buffer<T> *question_value = nullptr;
+  MLCommon::device_buffer<T> *temp_data = nullptr;
 
   //Total temp mem
   size_t totalmem = 0;
 
   //CUDA stream
   cudaStream_t stream;
 
-  //For quantiles
+  //No of SMs
+  int no_sms;
```

Comment:
knitpick: can you please rename this to `num_sms` to avoid confusion? It's easy to misread this is `no` SMs are being used :)

Reply:
done thanks